common/src/main/java/org/opensearch/ml/common/connector/functions/postprocess/ConnectorPostProcessFunction.java

```diff
@@ -6,22 +6,23 @@
 package org.opensearch.ml.common.connector.functions.postprocess;
 
 import java.util.List;
-import java.util.function.Function;
+import java.util.function.BiFunction;
 
+import org.opensearch.ml.common.output.model.MLResultDataType;
 import org.opensearch.ml.common.output.model.ModelTensor;
 
-public abstract class ConnectorPostProcessFunction<T> implements Function<Object, List<ModelTensor>> {
+public abstract class ConnectorPostProcessFunction<T> implements BiFunction<Object, MLResultDataType, List<ModelTensor>> {
```

Comment:
This interface change looks like a breaking change, and not all PostProcessFunctions are using the ResultDataType like BedrockBatchJobArnPostProcessFunction. So what I am thinking is 1. to use extend instead of implements to support additional parameter MLResultDataType, or 2. to have MLResultDataType covering all cases and all PostProcessFunctions to use/set it, for example, not use null as dateType but have a default type something like FLOAT32.

Reply:
I considered to change the Function to a self-defined interface/abstract class, but that needs more code changes and I want to limit the changes in this PR for easier reviewing. I'm open to make these changes in this PR if you insist or create a separate PR for better reviewing experience.

Reply:
Yeah, for this pr, I am OK. We can have more discussion about if we need refactoring later.